enginetest/transaction_queries.go

```diff
@@ -34,6 +34,87 @@ type TransactionTest struct {
 }
 
 var TransactionTests = []TransactionTest{
+	{
+		// Repro for https://github.com/dolthub/dolt/issues/3402
+		Name: "Changes from transactions are available before analyzing statements in other sessions (autocommit on)",
+		Assertions: []ScriptTestAssertion{
+			{
+				Query:    "/* client a */ select @@autocommit;",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:    "/* client b */ select @@autocommit;",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:       "/* client a */ select * from t;",
+				ExpectedErr: sql.ErrTableNotFound,
+			},
+			{
+				Query:       "/* client b */ select * from t;",
+				ExpectedErr: sql.ErrTableNotFound,
+			},
+			{
+				Query:    "/* client a */ create table t(pk int primary key);",
+				Expected: []sql.Row{{sql.OkResult{}}},
+			},
+			{
+				Query:    "/* client b */ select count(*) from t;",
+				Expected: []sql.Row{{0}},
+			},
+		},
+	},
+	{
```

Comment:
This test is probably more appropriate to define in dolt, as this is definitely not how DDL changes work in MySQL (they take a global lock and are not transactional)